operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/source/informer/InformerEventSource.java

```diff
@@ -11,87 +13,218 @@
 import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
 import io.javaoperatorsdk.operator.processing.event.Event;
 import io.javaoperatorsdk.operator.processing.event.EventHandler;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
 import io.javaoperatorsdk.operator.processing.event.source.ResourceCache;
 
-public class InformerEventSource<T extends HasMetadata, P extends HasMetadata>
-    extends ManagedInformerEventSource<T, P, InformerConfiguration<T, P>>
-    implements ResourceCache<T>, ResourceEventHandler<T> {
+public class InformerEventSource<R extends HasMetadata, P extends HasMetadata>
+    extends ManagedInformerEventSource<R, P, InformerConfiguration<R, P>>
+    implements ResourceCache<R>, ResourceEventHandler<R> {
+
+  private static final Logger log = LoggerFactory.getLogger(InformerEventSource.class);
 
-  private final InformerConfiguration<T, P> configuration;
+  private final InformerConfiguration<R, P> configuration;
+  private final EventBuffer<R> eventBuffer = new EventBuffer<>();
+  private final ReentrantLock lock = new ReentrantLock();
```

Comment:
AFAIK, synchronize is discouraged since ReentrantLock-s were introduced.

Reply:
My understanding was that locks should be used to decrease the size of the critical section but in these cases, the whole method is locked so maybe it would read better to just have the method synchronized? I don't think that there is a semantical difference in this situation (but I might be wrong).

Reply:
will change it